keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -338,3 +339,170 @@ parameters: string:string [8]
 1. `secretRef`: (Optional) The name and namespace of an existing secret to be copied to the `Bucket`'s namespace for static provisioning.  Requires that `bucketIdentifier` be defined. Used for brownfield and static cases.
 1. `parameters`: (Optional) Object store specific key-value pairs passed to the driver.
 
+### RPC
+
+```
+syntax = "proto3";
+
+package cosi.v1;
+
+import "google/protobuf/descriptor.proto";
+
+option go_package = "github.com/container-object-store-interface/go-cosi";
+
+extend google.protobuf.MessageOptions {
+    bool cosi_secret = 1000;
+}
+
+message DriverInfoRequest {} // INTENTIONALLY BLANK
+
+enum DataProtocol {
+    PROTOCOL_UNSPECIFIED = 0;
+    AZURE = 1;
+    GCS = 2;
+    S3 = 3;
+}
+
+enum AccessMode {
+    MODE_UNSPECIFIED = 0;
+    RO = 1;
```

Comment:
Does "RO" == read only?
Does "WO" == write only or write once?
"RW" seems easy to guess as read-write-many.

Reply:
the *Once and *Many suffixes are K8s terms for volumes (private vs shared).  So I don't think we should worry about confusing people between *Only and *Once.  Also, all buckets are *Many by nature to anyone on the network.

Reply:
Agree for k8s. But here we're defining the CO-agnostic COSI spec so we should not assume k8s vocabulary is familiar -- at least that's the theory...